src/consensus/tx_verify.cpp

```diff
@@ -149,6 +149,8 @@ unsigned int GetP2SHSigOpCount(const CTransactionRef &tx, const CCoinsViewCache
         for (unsigned int i = 0; i < tx->vin.size(); i++)
         {
             CoinAccessor coin(inputs, tx->vin[i].prevout);
+            assert(!coin.IsSpent());
+            const CTxOut &prevout = coin.out;
```

Comment:
the assert here should be changed to return some error instead of asserting because there is no indication that the tx needs to be verified before running this function. 

line 153 prevout variable is unused